stock_inventory_revaluation/models/stock_inventory_revaluation.py

```diff
@@ -0,0 +1,574 @@
+                self.revaluation_type == 'price_change':
+            raise UserError(_("Revaluation type 'Unit Price Change' doesn't "
+                              "apply to Stock-Move-Specific changes.  Use "
+                              "type 'Total Value Change'."))
+
+        if self.product_id.cost_method in ['standard', 'average']:
+            if self.revaluation_type == 'price_change':
+                if self.current_cost == self.new_cost:
+                    raise UserError(_("Change the cost before posting"))
+            else:
+                if self.current_value == self.new_value:
+                    raise UserError(_("Change the value before posting"))
+
+    @api.model
+    def _validate_fifo_moves(self):
+        if not self.reval_move_ids:
+            raise UserError(_("Please revalue one or more stock moves"))
+
+        prec = self.env['decimal.precision'].precision_get('Account')
+        abs_change = sum(
+            [abs(x.new_value - x.current_value) for x in self.reval_move_ids]
+        )
+        if float_compare(abs_change, 0.0, precision_digits=prec) == 0:
+            raise UserError(_("No changes were made to move values"))
+
+        move_products = self.reval_move_ids.mapped('product_id.id')
+        if not all(x == self.product_id.id for x in move_products):
+            raise UserError(_("The selected moves are for a different "
+                              "product than the revaluation"))
+
+    @api.model
+    def _set_validate_accounting_parameters(self):
+        """Set and validate accounting fields, in preparation to post"""
+        if not self.decrease_account_id or not self.increase_account_id:
+            raise UserError(
+                _("Please add an Increase Account and a Decrease Account"))
+        if not self.post_date:
+            self.post_date = fields.Date.today()
+        acc = self.product_template_id.get_product_accounts()
+        if not acc.get('stock_valuation'):
+            raise UserError(_("Please add a Stock Valuation Account in the "
+                              "Product Category"))
+        self.stock_valuation_account_id = acc['stock_valuation']
+        self.journal_id = self.journal_id or acc.get('stock_journal')
+        if not self.journal_id:
+            raise UserError(_("Please specify an accounting journal"))
+
+    @api.multi
+    def post(self):
+        for reval in self:
+            reval._validate_product_parameters()
+
+            if reval.product_id.cost_method in ['standard', 'average']:
+                reval.old_cost = reval.current_cost
+                reval.old_value = reval.current_value
+                if reval.revaluation_type == 'price_change':
+                    reval.new_value = reval.qty_available * reval.new_cost
+                    reval.product_id.with_context(
+                        force_company=reval.company_id.id
+                    ).sudo().write({'standard_price': reval.new_cost})
```

Comment:
I have some rounding issue, example:
Input: Product AVCO / Stock qty = 147 / Current cost = 3,40 / Current value = 499,80
Parameters: New value = 750
Output: Current value = 749,70

Reply:
The problem is in a core module: stock_account.  If accounting precision is 2 digits, then the standard_price field on the product gets rounded from 5.102041 to 5.10.  So, for standard and average cost, it doesn't matter whether the stock.move.remaining_value is 749.70 or 750.00.
product.stock_value = product.standard_price * product.qty_available
In this case, 5.1 * 147 = 749.70

Reply:
I have the Product Price decimal precision set much higher than the Account precision. For example, set Product Price to 6 digits, and leave Account at 2 digits. Then, current_cost will be 750.000027, and the journal entry will be for 750.00 - 499.80 = 250.20

Reply:
Leaving it the way it is, unless you have a solution in mind.